ts/Extensions/Annotations/Types/VerticalLine.ts

```diff
@@ -58,31 +59,34 @@ class VerticalLine extends Annotation {
         target: Highcharts.AnnotationControllable
     ): Highcharts.AnnotationMockPointOptionsObject {
         var annotation = target.annotation as Highcharts.AnnotationVerticalLine,
+            chart = annotation.chart,
+            inverted = chart.inverted,
             point = annotation.points[0],
-            xy = MockPoint.pointToPixels(point, true),
-            y = xy.y,
-            offset = annotation.options.typeOptions.label.offset;
-
-        if (annotation.chart.inverted) {
-            y = xy.x;
-        }
+            left = pick(point.series.yAxis?.left, 0),
+            top = pick(point.series.yAxis?.top, 0),
+            offset = annotation.options.typeOptions.label.offset,
+            y = MockPoint.pointToPixels(point, true)[inverted ? 'x' : 'y'];
 
         return {
             x: point.x as any,
             xAxis: point.series.xAxis,
-            y: y + offset
+            y: y + offset +
+                (inverted ? (left - chart.plotLeft) : (top - chart.plotTop))
```

Comment:
Personally, I wouldn't split that into two lines, but it's up to You ;)

Reply:
I like to keep 80 characters line length :) in this case it's readable enough so I would leave it as it is.